include/nbl/builtin/hlsl/shapes/triangle.hlsl

```diff
@@ -0,0 +1,116 @@
+
+#ifndef _NBL_BUILTIN_HLSL_SHAPES_TRIANGLE_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SHAPES_TRIANGLE_INCLUDED_
+
+#include <nbl/builtin/hlsl/common.hlsl>
+#include <nbl/builtin/hlsl/math/functions.hlsl>
+#include <nbl/builtin/hlsl/limits/numeric.hlsl>
+
+namespace nbl
+{
+namespace hlsl
+{
+
+namespace shapes
+{
+
+//
+float3x3 getSphericalTriangle(in float3x3 vertices, in float3 origin)
+{
+	float3x3 tp_vertices = transpose(vertices);
+
+    // the `normalize` cannot be optimized out
+    return float3x3(normalize(tp_vertices[0]-origin), normalize(tp_vertices[1]-origin), normalize(tp_vertices[2]-origin));
```

Comment:
check matrix math and who uses this, if this matrix was used as a plain `float3[3]` no problem, worse if people were multiplying things with it